web/src/settings_components.ts

```diff
@@ -976,6 +978,15 @@ export function populate_data_for_realm_settings_request(
                     });
                     continue;
                 }
+
+                if (property_name === "can_create_private_channel_group") {
+                    data[property_name] = JSON.stringify({
+                        new: input_value,
+                        old: realm.realm_can_create_private_channel_group,
+                    });
+                    continue;
+                }
```

Comment:
I think we should refactor this such that we do not add separate if blocks for each setting.

I tried doing something so we can just use a single if block by trying to compute old value like `old: realm["realm_" + property_name]`, but typescript raised error. Will look into this again.

Reply:
Agreed both on strategy and that this seems reasonable to do in a follow-up PR.